### PR TITLE
[GJ-41] Support WS Tansformer with row-based output

### PR DESCRIPTION
**Pull request by zzcclp**
1. Support ClickHouse as native engine;
2. Support Row iterator;
3. Add config 'spark.oap.sql.columnar.iterator' to specify whether to use columnar basic iterator

Following PR:
1. JNI Interface
2. Optimise Code 

Closes https://github.com/oap-project/gazelle-jni/issues/41.

### CODE REVIEW

File: jvm/src/main/java/com/intel/oap/vectorized/CHJniInstance.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
+/*
```
Comment (rui-mo) [inline jvm/src/main/java/com/intel/oap/vectorized/CHJniInstance.java:1]: Can we restructure the files? For example, the row related files can be put into a specific row folder, not in the **vectorized** foler. Also, the CH related codes can be reused by other libs as well, so I think their names and paths should not contain library related information.

Comment (zzcclp) [reply jvm/src/main/java/com/intel/oap/vectorized/CHJniInstance.java:1]: > Can we restructure the files? For example, the row related files can be put into a specific row folder, not in the **vectorized** foler. Also, the CH related codes can be reused by other libs as well, so I think their names and paths should not contain library related information.

Sure, we can restructure these files in the following PR, when JNI interfaces are consistent, this file isn't needed and will be removed.

Comment (zzcclp) [reply jvm/src/main/java/com/intel/oap/vectorized/CHJniInstance.java]: Done

**Comment by rui-mo** [comment]
This option **spark.oap.sql.columnar.iterator** and related logic are expected to be removed after Kyligence transfer to columnar computing.

**Review by rui-mo — APPROVED** [review]
LGTM

**Comment by rui-mo** [comment]
Verified this PR on both master and Velox branches.

**Pull request merged**